examples/tp/tm_high_order.py

```diff
@@ -0,0 +1,465 @@
+# where the TM doesn't output any prediction)
+print ""
+print "-"*50
+print "Part 1. We present the sequence ABCD to the TM. The TM will eventually"
+print "will learn the sequence and predict the upcoming characters. This can be"
+print "measured by the prediction accuracy in Fig 1."
+print "N.B. In-between sequences the accuracy is 0.0 as the TM does not output"
+print "any prediction."
+print "-"*50
+print ""
+
+x = []
+y = []
+trainTM(seq1, timeSteps=10, noiseLevel=0.0)
+
+plt.ylim([-0.1,1.1])
+plt.plot(x, y)
+plt.xlabel("Timestep")
+plt.ylabel("Prediction Accuracy")
+plt.title("Fig. 1: TM learns sequence ABCD")
+plt.savefig("figure_1")
+plt.close()
+
+print ""
+print "-"*50
+print "Once the TM has learned the sequence ABCD, we will present the individual"
+print "characters to the TM to know its prediction. The TM outputs the columns"
+print "that become active upon the presentation of a particular character as well"
+print "as the columns predicted in the next time step. Here, you should see that"
+print "A predicts B, B predicts C, C predicts D, and D does not output any"
+print "prediction."
+print "N.B. Here, we are presenting individual characters, that is, a character"
+print "deprived of context in a sequence. There is no prediction for characters"
+print "X and Y as we have not presented them to the TM in any sequence."
+print "-"*50
+print ""
+
+showPredictions()
+
+print ""
+print "-"*50
+print "Part 2. We now present the sequence XBCY to the TM. As expected, the accuracy will"
+print "drop until the TM learns the new sequence (Fig 2). What will be the prediction of"
+print "the TM if presented with the sequence BC? This would depend on what character"
+print "anteceding B. This is an important feature of high-order sequences."
+print "-"*50
+print ""
+
+x = []
+y = []
+trainTM(seq2, timeSteps=10, noiseLevel=0.0)
+
+# In this figure you can see how the TM starts making good predictions for particular
+# characters (spikes in the plot). Then, it will get half of its predictions right, which
+# correspond to the times in which is presented with character C. After some time, it
+# will learn correctly the sequence XBCY, and predict its characters accordingly.
+plt.ylim([-0.1,1.1])
+plt.plot(x, y)
+plt.xlabel("Timestep")
+plt.ylabel("Prediction Accuracy")
+plt.title("Fig. 2: TM learns new sequence XBCY")
+plt.savefig("figure_2")
+plt.close()
+
+print ""
+print "-"*50
+print "We will present again each of the characters individually to the TM, that is,"
+print "not within any of the two sequences. When presented with character A the TM"
+print "predicts B, B predicts C, but this time C outputs a simultaneous prediction of"
+print "both D and Y. In order to disambiguate, the TM would require to know if the"
+print "preceding characters were AB or XB. When presented with character X the TM"
+print "predicts B, whereas Y and D yield no prediction."
+print "-"*50
+print ""
+
+showPredictions()
```

Comment:
I don't see how this example works given that currently showPredictions doesn't do any resets.